packages/commonwealth/client/scripts/controllers/server/roles.ts

```diff
@@ -299,3 +307,61 @@ export class RolesController {
     return this.User.addresses.find((a) => a.id === role.address_id);
   }
 }
+
+// Client-side helpers
+export function isActiveAddressPermitted(
+  active_address_roles: RoleInfo[],
+  chain_info: ChainInfo,
+  action: Action
+): PermissionError | undefined {
+  const chainRoles = active_address_roles.filter(
+    (r) => r.chain_id === chain_info.id
+  );
+
+  if (chainRoles.length > 0) {
+    // sort roles by roles with lowest roles first
+    chainRoles.sort((a) => {
```

Comment:
All this logic is basically replicated -- any way we can make it into a helper, for consistency?

Reply:
I made a helper and stored in shared/utils so that the client and server can use it